src/backend/joanie/core/filters/client.py

```diff
@@ -209,11 +236,23 @@ class CourseProductRelationViewSetFilter(filters.FilterSet):
         choices=enums.PRODUCT_TYPE_CHOICES,
         exclude=True,
     )
+    query = filters.CharFilter(method="filter_by_query")
```

Comment:
This is a tricky one `product_title_or_organization_title_or_course_code` … atrocious …
So maybe something like `po_title_or_c_code` ?